proposals/NNNN-weak-let.md

```diff
@@ -0,0 +1,106 @@
+# Feature name
+
+* Proposal: [SE-NNNN](NNNN-weak-let.md)
+* Authors: [Mykola Pokhylets](https://github.com/nickolas-pohilets)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+* Implementation: [swiftlang/swift#80440](https://github.com/swiftlang/swift/pull/80440)
+* Upcoming Feature Flag: `WeakLet`
+* Review: ([discussion](https://forums.swift.org/t/weak-captures-in-sendable-sending-closures/78498))
+
+## Introduction
+
+Currently Swift requires weak stored variables to be mutable.
+This restriction is rather artificial, and causes friction with sendability checking.
+
+## Motivation
+
+Currently swift classes with weak stored properties cannot be `Sendable`,
+because weak properties have to be mutable, and mutable properties are 
+not allowed in `Sendable` classes.
+
+Similarly, closures with `weak` captures cannot be `@Sendable`,
+because such captures are implicitly made mutable.
+
+Usually developers are not aware of this implicit mutability and have no intention to modify the captured variable.
+Implicit mutability of weak captures is inconsistent with `unowned` or default captures.
+
+Wrapping weak reference into a single-field struct, allows stored properties and captures to be immutable.
+
+```swift
+final class C: Sendable {}
+
+struct WeakRef {
+    weak var ref: C?
+}
+
+final class User: Sendable {
+    weak let ref1: C? // error: 'weak' must be a mutable variable, because it may change at runtime
+    let ref2: WeakRef // ok
+}
+
+func makeClosure() -> @Sendable () -> Void {
+    let c = C()
+    return { [weak c] in
+        c?.foo() // error: reference to captured var 'c' in concurrently-executing code
+        c = nil // nobody does this
+    }
+    return { [c = WeakRef(ref: c)] in 
+        c.ref?.foo() // ok
+    }
+}
+```
+
+Existence of this workaround shows that ban on `weak let` variables is artificial, and can be lifted.
```

Comment:
I feel like the workaround is more likely to be seen as a bug than something to be extended.